internal/uvm/vnuma.go

```diff
@@ -0,0 +1,184 @@
+//go:build windows
+
+package uvm
+
+import (
+	"fmt"
+
+	hcsschema "github.com/Microsoft/hcsshim/internal/hcs/schema2"
+)
+
+// prepareVNumaTopology creates vNUMA settings for implicit (platform) or explicit (user-defined) topology.
+//
+// For implicit topology we look for `MaxProcessorsPerNumaNode` and `MaxSizePerNode` create options. Setting them
+// in HCS doc, will trigger platform to create vNUMA topology based on the given values. Based on experiments, the
+// platform will create an evenly distributed topology based on requested memory and processor count for the HCS VM.
+//
+// For explicit topology we look for `NumaMappedPhysicalNodes`, `NumaProcessorCounts` and `NumaMemoryBlocksCounts` create
+// options. The above options are number slices, where a value at index `i` in each slice represents the corresponding
+// value for the `i`th vNUMA node.
+// Limitations:
+// - only hcsschema.MemoryBackingType_PHYSICAL is supported
+// - `PhysicalNumaNodes` values at index `i` will be mapped to virtual node number `i`
+// - client is responsible for setting wildcard physical node numbers
+//
+// TODO: We also assume that `hcsschema.Numa.PreferredPhysicalNodes` can be used for implicit placement as well as
```

Comment:
What is left to do here?

Reply:
I have not validated this implicit nNUMA configuration. 
@anmaxvl: Is this comment w.r.t pending validation?

Reply:
right. I think we'd want to validate how this setting affects scenarios like:
1. Implicit placement + preferred physical nodes
2. Explicit placement with wildcard/invalid physical node numbers + preferred physical nodes
3. Explicit placement with partial wildcard/invalid physical node topology + preferred physical nodes

Reply:
I've confirmed that 1 and 2 work as one would expect: the platform will honor preferred physical nodes when creating an implicit vNUMA topology or when assigning pNUMA indices to wildcard/invalid physical node numbers.

3 is actually not a supported scenario and the validation code handles it.

That being said though, I think we should not support explicit placement + preferred physical nodes in the initial implementation and see if anyone would actually need this semi-implicit placement in the future. This means that we should return an error when:
- preferred physical nodes is used together with mapped physical nodes
- mapped physical nodes contain a wildcard/invalid node number (0xFF)